bionty/base/_public_ontology.py

```diff
@@ -351,7 +337,15 @@ def df(self) -> pd.DataFrame:
             bt_base.Gene().df()
         """
         if "ontology_id" in self._df.columns:
-            return self._df.set_index("ontology_id")
+            # Filter ontology_id by source prefix
+            df = self._df[
```

Comment:
I think the logic of this section can potentially be a bit simplified by updating the `self._df` and setting the index once at the end before returning it. Like we had before except that now we have the additional filtering step.